### PR TITLE
Updates comments in the openst-protocol repo [mosa-98]-II

### PR DESCRIPTION
**Pull request by noslav**
Files touched - 
1. BrandedToken.sol
2. Core.sol
3. CoreInterface.sol
4. EIP20Interface.sol
5. EIP20Token.sol
6. EIP20TokenMock.sol
7. Gateway.sol
8. Hasher.sol
9. OpsManaged.sol
10. Owned.sol
11. SafeMath.sol

Guidelines - 
* Replaces `///` document comment style with `/** */` one.
* One-liner doc comments follow the same guideline: /** */ -  for struct, enum and member vars.
* For multiline doc comments, it should be
`/**`
   `*  @title A title that should describe the contract or interface.`
   `*`
   `*  @notice Explain to a user what a contract, interface or function does.`
   `*`
   `*  @dev Explain to a developer any extra details for a contract, interface or function.`
   `*`
   `*  @param _firstParameterName Documents a parameter of a function`
   `*`              `just like in doxygen.`
   `*  @param _secondParamName Followed by second param details.`
   `*`
   `*  @return type Documents the return type of a contract's function.`
   `*/`
* All comment lines start with caps and end with a period, using proper grammar.

### CODE REVIEW

File: contracts/Core.sol [hunk 1]
```diff
@@ -71,26 +65,28 @@ contract Core is CoreInterface, Util {
...
 	bytes private encodedOpenSTRemotePath;
 
-	/* Modifiers */
+	/** Modifiers */
 
-	/** only worker modifier */
+    /**
+     *  @notice Modifier onlyWorker.
+     *
+     *  @dev Checks if msg.sender is whitelisted workder address to proceed.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Core.sol]: Done!

File: contracts/Core.sol [hunk 2]
```diff
@@ -28,41 +28,35 @@ import "./WorkersInterface.sol";
 import "./RLP.sol";
 
 /**
- *	@title Core contract which implements CoreInterface
+ *	@title Core contract which implements CoreInterface.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Core.sol]: Done!

File: contracts/CoreInterface.sol [hunk 3]
```diff
@@ -28,23 +28,21 @@ pragma solidity ^0.4.23;
  */
 contract CoreInterface {
 
-	function registrar() public view returns (address /* registrar */);
+	/** Public Functions */
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/CoreInterface.sol]: Done!

File: contracts/EIP20Token.sol [hunk 4]
```diff
@@ -95,7 +149,18 @@ contract EIP20Token is EIP20Interface {
...
-
+    /**
+     *  @notice Public function approve.
+     *
+     *  @dev Allows _spender address to withdraw from function caller's account, multiple times up 
+     *       to the _value amount, if this function is called again 
+     *       it overwrites the current allowance with _value.
+     *
+     *  @param _spender Address authorized to spend from the function caller's address.
+     *  @param _value Amount upto which spender is authorized to spend.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/EIP20Token.sol]: Done!

File: contracts/EIP20TokenMock.sol [hunk 5]
```diff
@@ -23,20 +23,51 @@ pragma solidity ^0.4.23;
...
         return 0;
     }
 
-    /// @dev Public wrapper for claimEIP20Public
+    /**
+     *  @notice Public function claimEIP20Public.
+     * 
+     *  @dev Public wrapper for claimEIP20.
+     *
+     *  @param _beneficiary Address of tokens beneificary.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/EIP20TokenMock.sol]: Done!

File: contracts/EIP20TokenMock.sol [hunk 6]
```diff
@@ -55,7 +94,15 @@ contract EIP20TokenMock is EIP20Token {
         return mintEIP20(_amount);
     }
 
-    /// @dev Public wrapper for burnEIP20Public
+    /**
+     *  @notice Public function burnEIP20Public
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/EIP20TokenMock.sol]: Done!

File: contracts/EIP20TokenMock.sol [hunk 7]
```diff
@@ -55,7 +94,15 @@ contract EIP20TokenMock is EIP20Token {
...
+     * 
+     *  @dev Public wrapper for burnEIP20
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/EIP20TokenMock.sol]: Done!

File: contracts/Hasher.sol [hunk 8]
```diff
@@ -21,10 +21,28 @@ pragma solidity ^0.4.23;
...
+	/**
+	 *	@notice Public pure function 
+	 *	
+	 *	@param _symbol Symbol of the token.
+	 *	@param _name Name of the token.
+	 *	@param _chainIdValue Chain id of the value chain.
+	 *	@param _chainIdUtility Chain id of the utility chain.
+	 *	@param _openSTUtility Address of the openSTUtility contract.
+     *  @param _conversionRate Conversion rate of the token.
+     *  @param _conversionRateDecimals Decimal places of conversion rate of token.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Hasher.sol]: Done!

File: contracts/Hasher.sol [hunk 9]
```diff
@@ -74,6 +106,19 @@ contract Hasher {
 				_hashLock));
 	}
 
+	/**
+	 *	@notice Public pure function 
+	 *	
+	 *	@param _uuid Uuid of the token.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Hasher.sol]: Done!

File: contracts/Hasher.sol [hunk 10]
```diff
@@ -49,6 +67,20 @@ contract Hasher {
...
+	 *	@param _account Address of the staking account.
+	 *	@param _accountNonce Nonce of the staking account.
+	 *	@param _beneficiary Address of the beneficiary on utility chain.
+	 *	@param _amountST Amount of ST staked.
+	 *	@param _amountUT Amount of UT to mint.
+	 *	@param _unlockHeight Block height upto which staking is locked.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Hasher.sol]: Done!

File: contracts/Hasher.sol [hunk 11]
```diff
@@ -21,10 +21,28 @@ pragma solidity ^0.4.23;
...
+ *			required for the process of stake and mint / redeem and unstake.
+ */
 contract Hasher {
 
-	/*
-	 *  Public pure functions
+	/**  Public functions */
+
+	/**
+	 *	@notice Public pure function
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Hasher.sol]: @notice Public pure function This one should be the consistent version that should exist upon update.

File: contracts/BrandedToken.sol [hunk 12]
```diff
@@ -24,26 +24,43 @@ pragma solidity ^0.4.23;
...
+ *         their equivalent part of the Simple Token stake
+ *         on Ethereum (before v1.0).
+ */
 contract BrandedToken is EIP20Token, UtilityTokenAbstract {
     using SafeMath for uint256;
 
-    /*
-     *  Public functions
+    /**
+     *  @notice Contract Constructor.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/BrandedToken.sol]: Lowercase done!

File: contracts/Core.sol [hunk 13]
```diff
@@ -71,28 +65,30 @@ contract Core is CoreInterface, Util {
...
-	 * @dev bytes32ToBytes is util contract method
+	 *  @dev bytes32ToBytes is util contract method.
 	 *
-	 * @param _registrar address of the registrar which registers for utility tokens
-	 * @param _chainIdOrigin chain id where current core contract is deployed since core contract can be deployed on remote chain also
-	 * @param _chainIdRemote if current chain is value then _chainIdRemote is chain id of utility chain
-	 * @param _openSTRemote if current chain is value then _openSTRemote is address of openSTUtility contract address
-	 * @param _blockHeight block height at which _stateRoot needs to store
-	 * @param _stateRoot state root hash of given _blockHeight
+	 * @param _registrar Address of the registrar which registers for utility tokens.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Core.sol]: Done!

File: contracts/EIP20TokenMock.sol [hunk 14]
```diff
@@ -23,20 +23,51 @@ pragma solidity ^0.4.23;
...
+
+/**
+ *  @title EIP20TokenMock contract.
+ *
+ *  @notice Implements mock totalSupply function and wraps internal 
+ *          functions to enable testing EIP20Token.
+ */
 contract EIP20TokenMock is EIP20Token {
 
+    /** Public Functions */
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/EIP20TokenMock.sol]: Done!

File: contracts/Hasher.sol [hunk 15]
```diff
@@ -97,6 +142,14 @@ contract Hasher {
 				_hashLock));
 	}
 
+	/**
+	 *  @notice Public pure function.
+	 *
+	 *  @param _account Address of the hashing account.
+	 *  @param _nonce Nonce of the hashing account.
+	 *
+	 *.  @return bytes32 Keccak256 intent key hash.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Hasher.sol]: Done!

File: contracts/Hasher.sol [hunk 16]
```diff
@@ -21,10 +21,28 @@ pragma solidity ^0.4.23;
 //
 // ----------------------------------------------------------------------------
 
+/**
+ *  @title Hasher contract. 
+ *
+ *  @notice Hasher contains functions for hashing frequently occuring state variables
+ *.         required for the process of stake and mint / redeem and unstake.
```
[the remark this answers is not in the document]
Comment (noslav) [reply contracts/Hasher.sol]: Done!

**Comment by jasonklein** [comment]
LGTM

**Pull request merged**